custom_components/tuya_local/devices/rainpoint_water_timer.yaml

```diff
@@ -0,0 +1,116 @@
+name: RainPoint Smart Water Timer
+products:
+  - id: ew946yrp3pgbaziu
+    name: RainPoint TTV103FRF
+primary_entity:
+  entity: valve
+  class: water
+  icon: "mdi:watering-can"
+  dps:
+    - id: 108
+      type: boolean
+      name: valve
+      mapping:
+        - dps_val: true
+          value: false
+        - dps_val: false
+          value: true
+secondary_entities:
+  - entity: switch
+    name: Weekly Automatic Irrigation
+    category: config
+    icon: "mdi:calendar-clock"
+    dps:
+      - id: 125
+        optional: true
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+            value: false
+          - dps_val: false
+            value: true
+  - entity: select
+    name: Weekly Irrigation Rain Delay
+    category: config
+    icon: "mdi:timer-sand"
+    dps:
+      - id: 113
+        optional: true
+        type: string
+        name: option
+        mapping:
+          - dps_val: "0"
+            value: "0h"
+          - dps_val: "24"
+            value: "24h"
+          - dps_val: "48"
+            value: "48h"
+          - dps_val: "72"
+            value: "72h"
+  - entity: sensor
+    class: battery
+    category: diagnostic
+    dps:
+      - id: 105
+        type: string
+        name: sensor
+        unit: "%"
+        mapping:
+          - dps_val: "0"
+            value: 10
+          - dps_val: "1"
+            value: 50
+          - dps_val: "2"
+            value: 100
+  - entity: sensor
+    name: Water flow
+    class: water
+    category: diagnostic
+    icon: "mdi:water"
+    dps:
+      - id: 103
+        name: sensor
+        type: integer
+        unit: L
+  - entity: number
+    translation_key: timer
+    category: config
+    dps:
+      - id: 107
+        type: integer
+        name: value
+        unit: min
+        range:
+          min: 0
+          max: 60
+  - entity: sensor
+    name: Status
+    class: enum
+    icon: "mdi:list-status"
+    category: diagnostic
+    dps:
+      - id: 106
+        type: string
+        name: sensor
+        mapping:
+          - dps_val: "0"
+            value: "Off"
+          - dps_val: "1"
+            value: "Manual"
+          - dps_val: "2"
+            value: "Auto"
+          - dps_val: "3"
+            value: "Rain Delay"
+  - entity: binary_sensor
+    category: diagnostic
+    class: problem
+    icon: "mdi:alert"
```

Comment:
problem sensor does not need an icon defined, the base class defines dynamic icons already